fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,281 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""Utility class for plotting metrics with error ranges."""
+
+from typing import List, Union
+import pandas as pd
+import numpy as np
+
+from ._metric_frame import MetricFrame
+from matplotlib.lines import Line2D
+from matplotlib.axes._axes import Axes
+
+_GIVEN_BOTH_ERRORS_AND_CONF_INT_ERROR = \
+    "Ambiguous Error Metric. Please only provide one of: errors or conf_intervals."
+_METRIC_AND_ERRORS_NOT_SAME_LENGTH_ERROR = \
+    "The list of metrics and list of errors are not the same length."
+_METRIC_LENGTH_ZERO_ERROR = \
+    "No metrics were provided to plot. A nonzero number of metrics is required."
+_METRIC_AND_CONF_INTERVALS_NOT_SAME_LENGTH_ERROR = \
+    "The list of metrics and list of conf_intervals are not the same length."
+
+_METRICS_NOT_LIST_OR_STR_ERROR = \
+    """Metric should be a string of a single metric or a list of the metrics
+     in provided MetricFrame, but {0} was provided"""
+_ERRORS_MUST_BE_SCALAR_OR_ARRAY = "Calculated errors must be a scalar or a array of length 2"
+_CONF_INTERVALS_MUST_BE_ARRAY = "Calculated conf_intervals must be array of length 2"
+_ERRORS_NEGATIVE_VALUE_ERROR = "Calculated relative errors must be positive"
+_CONF_INTERVALS_FLIPPED_BOUNDS_ERROR = \
+    "Calculated conf_intervals' upper bound cannot be less than lower bound"
+
+
+def _is_arraylike(input_):
+    return (isinstance(input_, np.ndarray) or isinstance(input_, list))
+
+
+def _check_if_metrics_and_error_metrics_same_length(metrics, errors, conf_intervals):
+    if errors is not None:
+        if len(errors) != len(metrics):
+            raise ValueError(_METRIC_AND_ERRORS_NOT_SAME_LENGTH_ERROR)
+    elif conf_intervals is not None:
+        if len(conf_intervals) != len(metrics):
+            raise ValueError(_METRIC_AND_CONF_INTERVALS_NOT_SAME_LENGTH_ERROR)
+
+
+def _check_if_metrics_length_zero(metrics):
+    if len(metrics) == 0:
+        raise ValueError(_METRIC_LENGTH_ZERO_ERROR)
+
+
+def _check_for_ambiguous_error_metric(errors, conf_intervals):
+    # ambiguous error metric if both are provided
+    if errors is not None and conf_intervals is not None:
+        raise ValueError(_GIVEN_BOTH_ERRORS_AND_CONF_INT_ERROR)
+
+
+def _check_for_valid_metrics_format(metrics):
+    # ensure metrics is either list, str, or None
+    if not (isinstance(metrics, list) or isinstance(metrics, str) or metrics is None):
+        raise ValueError(_METRICS_NOT_LIST_OR_STR_ERROR.format(type(metrics)))
+
+
+def _convert_scalar_error_to_array_error(df_errors):
+    if all((isinstance(error, float) or isinstance(error, int)) for error in df_errors):
+        return df_errors.apply(lambda x: [x, x])
+    if all((_is_arraylike(error) and len(error) == 2) for error in df_errors):
+        return df_errors
+    else:
+        raise ValueError(_ERRORS_MUST_BE_SCALAR_OR_ARRAY)
```

Comment:
I think this wouldn't be needed if we do confidence intervals only

Reply:
`df.plot` takes an error input, so at some point the confidence intervals need to be converted